agents/README.md

```diff
@@ -0,0 +1,223 @@
+# Reinforcement Learning with [tensorflow/agents](https://github.com/tensorflow/agents)
+
+[![Docker Repository on Quay](https://quay.io/repository/cwbeitel/agents-demo/status "Docker Repository on Quay")](https://quay.io/repository/cwbeitel/agents-demo)
+
+Here we provide a demonstration of training a reinforcement learning agent to perform a robotic grasping task using Kubeflow running on Google Kubernetes Engine. In this demonstration you will learn how to paramaeterize a training job, submit it to run on your cluster, monitor the job including launching a tensorboard instance, and finally producing renders of the agent performing the robotic grasping task.
+
+For clarity and fun you can check out what the product of this tutorial will look like by clicking through the render screenshot below to a short video of a trained agent performing a simulated robotic block grasping task:
+
+[![](demo/render_preview.png)](https://youtu.be/0X0w5XOtcHw)
+
+Below are instructions to first set up various preliminaries then launch the example notebook in a variety of forms.
+
+### Setup
+
+##### GCP and Kubeflow configuration
```

Comment:
My suggestion would be to have separate sections for Kubeflow and Cloud setup.

The Kubeflow section is where you'd describe setting up Argo (ideally that would just be a link to the relevant section in the user guide).

In the Cloud section, one subsection would be for GCP.
Could we have another subsection for other clouds and briefly mention what would need to be done?

I suspect the only thing that's needed is to setup object store to write the files?

Reply:
Ah ok I don't know how to do this with other cloud providers but I can figure that out. But yeah generally if once the kube cluster is created we keep everything within cluster then things are much simpler to set up.

If we use NFS in-cluster then we don't have to coordinate working with various vendors' object store interfaces and setup. Logging to NFS instead of GCS would also simplify the credential setup steps for TensorBoard.

Also there could be an addition made to the kubeflow deployment step, separately from this demo, wherein credentials are added to the user's notebook environment to interact with the underlying kube cluster.